### PR TITLE
Add dedicated test for chain replication

### PR DESCRIPTION
**Pull request by jasontedor**
This commit adds a dedicated test that chain replication leader -> middle -> follow is successful.

### CODE REVIEW

File: x-pack/plugin/ccr/qa/chain/src/test/java/org/elasticsearch/xpack/ccr/ChainIT.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                assertBusy(() -> verifyDocuments(middleIndexName, numDocs + 6, middleClient));
+            }
+
+            assertBusy(() -> verifyDocuments(followIndexName, numDocs + 6));
+        } else {
+            fail("unexpected target cluster [" + targetCluster + "]");
+        }
+    }
+
+    private static void index(RestClient client, String index, String id, Object... fields) throws IOException {
```
Comment (martijnvg) [inline x-pack/plugin/ccr/qa/chain/src/test/java/org/elasticsearch/xpack/ccr/ChainIT.java]: Should we in a follow up add a Gradle module inside the qa directory that contains these helper methods and let the qa modules depend on this module? We've been repeating these methods many times now.

**Review by martijnvg — APPROVED** [review]
👍 great qa test!

File: x-pack/plugin/ccr/qa/chain/src/test/java/org/elasticsearch/xpack/ccr/ChainIT.java (hunk 1 above)
Comment (jasontedor) [reply x-pack/plugin/ccr/qa/chain/src/test/java/org/elasticsearch/xpack/ccr/ChainIT.java]: Yes, I thought the same thing! Can you take that on?

Comment (martijnvg) [reply x-pack/plugin/ccr/qa/chain/src/test/java/org/elasticsearch/xpack/ccr/ChainIT.java]: 👍  I will work on it.

**Pull request merged**